lib/run_part.c

```diff
@@ -40,7 +41,7 @@ static int run_part(char *script_path, const char *name, const char *action)
 		return (wait_status);
 	}
 
-	fprintf(shadow_logfd, "waitpid: %s\n", strerror(errno));
+	fprintf(shadow_logfd, "failed to wait for pid %d (%s): %s\n", pid, script_name, strerror(errno));
```

Comment:
I'd also change this to a format `"wait(...): %s\n"` or something like that for consistency.